sklearn/utils/validation.py

```diff
@@ -498,10 +498,10 @@ def check_X_y(X, y, accept_sparse=None, dtype="numeric", order=None, copy=False,
                     ensure_2d, allow_nd, ensure_min_samples,
                     ensure_min_features, warn_on_dtype, estimator)
     if multi_output:
-        y = check_array(y, 'csr', force_all_finite=True, ensure_2d=False,
-                        dtype=None)
+        y = check_array(y, 'csr', copy=copy, force_all_finite=True,
+                        ensure_2d=False, dtype=None)
```

Comment:
How about `copy`? Right now I might need to have to use code like this.

    X, y = check_X_y(X, y, copy=False)
    std_scaler = StandardScaler(copy=False)
    std_scaler.fit_transform(X)
    std_scaler.copy = True
    std_scaler.fit_transform(y)

Though this is not a huge issue since y is copied only once anyhow. It might make the code look slightly cleaner.